pkgs/development/compilers/ghc/8.10.7.nix

```diff
@@ -156,6 +156,10 @@ stdenv.mkDerivation (rec {
     # upstream patch. Don't forget to check backport status of the upstream patch
     # when adding new GHC releases in nixpkgs.
     ./respect-ar-path.patch
+
+    # In order to build ghcjs packages, the Cabal of the ghc used for the ghcjs
+    # needs to be patched. Ref https://github.com/haskell/cabal/pull/7575
+    ./cabal-ghcjs-support.patch
```

Comment:
Is it possible to `fetchpatch` one of the commits from this PR (or possibly all commits from this PR) instead of adding this patch to nixpkgs?

In general, we try not to add raw .patch files to nixpkgs if we can avoid it.

Reply:
Unfortunately in this case it is not possible (or perhaps I dont know how to do) as the patch is for a submodule.

Reply:
I'm not sure I understand.  Is there no upstream commit with this patch's contents?

Reply:
Yes there is a commit, but it is in Cabal package, and in the ghc code it is present in "libraries/Cabal" path, so applying patch will not work. (unless we patch the paths in the patch)

Reply:
`fetchpatch` has options for changing paths in a `.patch` file.  Will any of them help you here?

Reply:
fixed